### PR TITLE
feat(YouTube): ReturnYouTubeDislike compact like button layout

### PR DESCRIPTION
**Pull request by LisoUseInAIKyrios**
RE: https://github.com/revanced/revanced-integrations/pull/284

Added an option for a slightly more narrow like button layout, that removes the left separator and tightens the space between the like/dislike by a few more pixels.

Mostly intended for users of devices with strange system fonts (such as [Xiaomi](https://github.com/revanced/revanced-integrations/issues/276)), or any other device where the left separator does not style well.

compact layout

full size

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
Another addition that could solve the issue with styling for all users is to add options that allow changing the style to the settings.

File: src/main/resources/returnyoutubedislike/host/values/strings.xml [hunk 1]
```diff
@@ -17,6 +16,10 @@
     <string name="revanced_ryd_dislike_percentage_summary_on">Dislikes shown as percentage</string>
     <string name="revanced_ryd_dislike_percentage_summary_off">Dislikes shown as number</string>
 
+    <string name="revanced_ryd_compact_layout_title">Compact like button layout</string>
+    <string name="revanced_ryd_compact_layout_summary_on">Showing narrow like button</string>
+    <string name="revanced_ryd_compact_layout_summary_off">Showing normal like button</string>
```
Comment (oSumAtrIX) [inline src/main/resources/returnyoutubedislike/host/values/strings.xml:21]: These do not convey what this switch does properly. Instead, it should say, that it makes it compact by getting rid of the left separator.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by LisoUseInAIKyrios** [comment]
I think letting users make adjustments to the style layout is good.  Would need to figure out how to present that to the user in a way that's not confusing or too clunky.

File: src/main/resources/returnyoutubedislike/host/values/strings.xml (hunk 1 above)
Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml:21]: It does describe what happens: the likes button becomes more narrow.  

The user may not realize what the left separator is.
They'll figure it out when they try turning it on and off.

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml:21]: maybe something like:

"Showing minimal like button layout"
"Showing sparse like button layout"

But I feel that 'narrow' describes what happens better.

Maybe some other suggestions?

**Comment by oSumAtrIX** [comment]
I'd go with a sub-preference screen with input boxes for inputting offsets and absolute values, like the `custom-buffer` patch.

File: src/main/resources/returnyoutubedislike/host/values/strings.xml (hunk 1 above)
Comment (oSumAtrIX) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml:21]: My following suggestion should be formulated/worded better, but something like this maybe:
```suggestion
    <string name="revanced_ryd_compact_layout_title">Compact like button text</string>
    <string name="revanced_ryd_compact_layout_summary_on">Only showing like and dislike separately</string>
    <string name="revanced_ryd_compact_layout_summary_off">Fully formatting like and dislike symmetrically</string>
```

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml:21]: maybe:
Compact like button
Like and Dislike displayed together
Like and Dislike displayed separately

Comment (oSumAtrIX) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml:21]: > Like and Dislike displayed together

But they are not displayed together at all are they?

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: I think this describes it well, and it reads ok too.

separately

together

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: Although I'll change it to so "Like**s** and Dislike**s**"

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: > > Like and Dislike displayed together
> 
> But they are not displayed together at all are they?

yeah I guess they are not.  It's "more together" but they still have the • separator between them.

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: Maybe:
Likes and Dislikes minimally styled
Likes and Dislikes fully styled

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: fully styled


minimally styled

Comment (oSumAtrIX) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: Tough. Maybe 

Likes/ Dislikes buttons styled symmetrically
Likes/ Dislikes button occupy a minimum amount of width

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: "occupy a minimum amount of width" is a bit wordy and might be confusing.

How about:

"Likes and Dislikes styled minimally"
"Likes and Dislikes styled symmetrically"

Comment (oSumAtrIX) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: Following on that suggestion, maybe this way:

"Likes and Dislikes styled for minimal width"
"Likes and Dislikes styled for best appearance"

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: yes, I like that best

Comment (LisoUseInAIKyrios) [reply src/main/resources/returnyoutubedislike/host/values/strings.xml]: best appearance

minimal width


I think this looks best

**Comment by oSumAtrIX** [comment]
Then the font of the like button span will look different from the system font.

**Comment by LisoUseInAIKyrios** [comment]
@YT-Advanced 
The issue is not so much the type of font, but instead the system default font size.

Most android phones use the default Android 'Roboto' font.  The exceptions to this include Google Pixel and a lot of Chinese manufacturers (Xiaomi, Oppo, OnePlus, etc).

Using a Google Pixel phone, I tried setting the entire like/dislike span to Roboto font, and then used the same segmented layout adjustments from Samsung (which also uses Roboto).

This did not work well.

Pixel phone:
Roboto font mixed with Google Sans font

Samsung phone:
Samsung phone

For the Pixel phone, the font of the 'likes' button visually looks the same as the other buttons.  Roboto is not very different from Google Sans, but with a different system font it might be noticeable.

Using the same font type does not make the layout adjustments universal.  Trying the same layout adjustments, these two devices had a different visual appearance, and it ends up about the same as a different no-name device running the currently released code (which does not change the font).

So unfortunately this won't help.

**Review by oSumAtrIX — APPROVED** [review]
